v12/storage-dotnet-redundancy-ha-ra-grs/Program.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System.Diagnostics.Tracing;
+using System.IO;
+using Azure;
+using Azure.Core;
+using Azure.Core.Diagnostics;
+using Azure.Core.Pipeline;
+using Azure.Identity;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace StorageRedundancy
+{
+    internal class ShowRequestInfoPolicy : HttpPipelinePolicy
+    {
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            if (host is not null) 
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            
+            ProcessNext(message, pipeline);
+        }
+
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string? host = message.Request.Uri.Host;
+            
+            if (host is not null)
+            {
+                string region = host.Contains("-secondary") ? "secondary" : "primary";
+                Console.WriteLine($"Request sent to {region} region endpoint");
+            }
+            await ProcessNextAsync(message, pipeline);
+
+            
+        }
+    }
+
+    public class Program
+    {
+        static int retryCount = 0;
+        const int retryThreshold = 3;
+        static int secondaryReadCount = 0;
+        const int secondaryReadThreshold = 5;
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Azure Storage redundancy sample\n ");
+
+            try
+            {
+                RunStorageRedundancyAsync().GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+
+        }
+
+        /// <summary>
+        /// Main method -> sets up the objects needed, then loops over blob download operations against primary and secondary endpoints
+        /// </summary>
+        private static async Task RunStorageRedundancyAsync()
+        {
+            // TODO: update accountName const to match your storage account name
+            const string accountName = "YOURSTORAGEACCOUNTNAME";
+            Uri primaryAccountUri = new Uri($"https://{accountName}.blob.core.windows.net/");
+            Uri secondaryAccountUri = new Uri($"https://{accountName}-secondary.blob.core.windows.net/");
```

Comment:
```suggestion
            Uri primaryAccountUri = new($"https://{accountName}.blob.core.windows.net/");
            Uri secondaryAccountUri = new($"https://{accountName}-secondary.blob.core.windows.net/");
```

Since you're being explicit about your type declarations, you can use the implicit new feature where you omit the type when instantiating the object.